ga4gh/cli.py

```diff
@@ -816,3 +820,72 @@ def client_main(parser=None):
     else:
         runner = args.runner(args)
         runner.run()
+
+##############################################################################
+# Configuration testing
+##############################################################################
+
+
+class TestConfigExists(unittest.TestCase):
+    """This class implements configuration checks for the GA4GH server. It is
+    structured as a set of unit tests, and relies on the unittest module.
+    However, as some of the tests require state injection (through command
+    line arguments) we need to break the usual unittest pattern and instead
+    implement our own simple test runner.
+    """
+
+    def __init__(self, parser=None):
+        self.parser = parser
+
+    def setUp(self):
+        self.app = flask.Flask(__name__)
+        parser = self.parser if self.parser is not None \
+            else argparse.ArgumentParser(
+                description="GA4GH server configuration validator")
+        parser.add_argument(
+            "--config", "-c", default='DevelopmentConfig', type=str,
+            help="The configuration to use")
+        parser.add_argument(
+            "--config-file", "-f", type=str, default=None,
+            help="The configuration file to use")
+        args = parser.parse_args()
+        config_str = 'ga4gh.serverconfig:{0}'.format(args.config)
+        self.app.config.from_object(config_str)
+        if os.environ.get('GA4GH_CONFIGURATION') is not None:
+            self.app.config.from_envvar('GA4GH_CONFIGURATION')
+        if args.config_file is not None:
+            self.app.config.from_pyfile(args.config_file)
+
+    def test_config_exists(self):
+        self.assertIsInstance(self.app.config["REQUEST_VALIDATION"],
+                              types.BooleanType,
+                              'REQUEST_VALIDATION is not a boolean')
+        self.assertIsNotNone(self.app.config["DATA_SOURCE"],
+                             'DATA_SOURCE must be set')
+
+
+def run_test(test_obj, test):
+    """
+    :param test_obj: The object under test
+    :param test: The test method to run
+    :return: Either None for success, or a tuple of the test name and raised
+    exception
+    """
+    try:
+        getattr(test_obj, test)()
+    except AssertionError as e:
+        return test, e
+    else:
+        return None
+
+
+def configtest_main(parser=None):
+    test_config = TestConfigExists(parser)
+    test_config.setUp()
+    tests = ['test_config_exists']
+    results = map(lambda t: run_test(test_config, t), tests)
+    print('{0}/{1} tests passed'.format(results.count(None), len(results)))
```

Comment:
we shouldn't have any `print` calls in the ga4gh module code ... these should be calls to some kind of logging tool (even if it is something as simple as calling a method that just calls print)